src/main/scala/org/broadinstitute/hail/Utils.scala

```diff
@@ -568,6 +589,133 @@ class RichPairTraversableOnce[K, V](val t: TraversableOnce[(K, V)]) extends AnyV
   }
 }
 
+
+class KeySortIterator[T, K, V](it: Iterator[(K, V)])(implicit tOrd: Ordering[T], kOrd: Ordering[K],
+  ev: (K) => T) extends Iterator[(K, V)] {
+
+  val b = mutable.ArrayBuffer[(K, V)]()
+  b.clear()
```

Comment:
Perhaps this is another instance of me misunderstanding Scala, but if we allocate a fresh buffer on line 596, would the buffer be non-empty on the following line?